### PR TITLE
fix: deactivate filter apply button when toggling unapplied filters

### PR DESCRIPTION
**Pull request by iskounen**
The type of this PR is: Bugfix

This PR resolves [FX-2870]

### Description

When a user selects options in a multi-select filter (ie. Medium, Rarity, Time period, Color) and un-selects those options without applying them, the "Show results" button should be de-activated. However, it currently remains active so this PR addresses that behavior.

#### Before

before

#### After

after

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [ ] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434
[FX-2870]: https://artsyproduct.atlassian.net/browse/FX-2870

### CODE REVIEW

File: src/lib/Components/ArtworkFilter/ArtworkFilterStore.tsx [hunk 1]
```diff
@@ -166,7 +166,7 @@ export const ArtworkFiltersModel: ArtworkFiltersModel = {
         ) {
           return true
         }
-        return defaultFilterOptions[paramName] !== paramValue
+        return !isEqual(defaultFilterOptions[paramName], paramValue)
```
Comment (iskounen) [inline src/lib/Components/ArtworkFilter/ArtworkFilterStore.tsx:169]: This was needed because `[] === []` is `false`.

Comment (MounirDhahri) [reply src/lib/Components/ArtworkFilter/ArtworkFilterStore.tsx:169]: ⭐️

**Review by MounirDhahri — APPROVED** [review]

File: src/lib/Components/ArtworkFilter/ArtworkFilterStore.tsx (hunk 1 above)
Comment (iskounen) [reply src/lib/Components/ArtworkFilter/ArtworkFilterStore.tsx:169]: @MounirDhahri just for your information, I made the same change to `applyFiltersAction` to fix the following bug where the count on the "Sort & Filter" pill was not being updated when applied filters were deselected.

pill

**Review by dzucconi — APPROVED** [review]
🙏

**Pull request merged**